lib/elixir/lib/calendar/date.ex

```diff
@@ -448,8 +460,16 @@ defmodule Date do
   @doc """
   Similar to `Date.convert/2`, but raises an `ArgumentError`
   if the conversion between the two calendars is not possible.
+
+  ## Examples
+
+  Imagine someone implements `Calendar.Julian`:
+
+      iex> Date.convert!(~D[2000-01-01], Calendar.Julian)
+      %Date{calendar: Calendar.Julian, year: 1999, month: 12, day: 19}
+
   """
-  @spec convert!(Date.t, Calendar.calendar) :: Date.t
+  @spec convert!(Calendar.date, Calendar.calendar) :: t | no_return
```

Comment:
We don't need `no_return`. `no_return` should be used only in clauses that always fail. Everything else is assumed that it may not return.